File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/playback/engine/selection/GivenAPlaybackEngineTypeSelectionBroadcaster/WhenBroadcastingTheEngineType.java
package com.lasthopesoftware.bluewater.client.playback.engine.selection.GivenAPlaybackEngineTypeSelectionBroadcaster;

import androidx.test.core.app.ApplicationProvider;

import com.lasthopesoftware.AndroidContext;
import com.lasthopesoftware.bluewater.client.playback.engine.selection.PlaybackEngineType;
import com.lasthopesoftware.bluewater.client.playback.engine.selection.broadcast.PlaybackEngineTypeChangedBroadcaster;
import com.lasthopesoftware.resources.FakeMessageBus;

import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class WhenBroadcastingTheEngineType extends AndroidContext {

	private static final FakeMessageBus fakeMessageSender = new FakeMessageBus(ApplicationProvider.getApplicationContext());

	@Before
	public void before() {
		new PlaybackEngineTypeChangedBroadcaster(fakeMessageSender)
			.broadcastPlaybackEngineTypeChanged(PlaybackEngineType.ExoPlayer);
	}

	@Test
	public void thenTheExoPlayerSelectionIsBroadcast() {
		assertThat(fakeMessageSender.getRecordedIntents().stream().findFirst().get().getStringExtra(PlaybackEngineTypeChangedBroadcaster.playbackEngineTypeKey)).isEqualTo(PlaybackEngineType.ExoPlayer.name());
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/playback/engine/selection/GivenAPlaybackEngineTypeSelectionBroadcaster/WhenBroadcastingTheEngineType.kt
package com.lasthopesoftware.bluewater.client.playback.engine.selection.GivenAPlaybackEngineTypeSelectionBroadcaster

import com.lasthopesoftware.bluewater.client.playback.engine.selection.PlaybackEngineType
import com.lasthopesoftware.bluewater.client.playback.engine.selection.broadcast.PlaybackEngineTypeChangedBroadcaster
import com.lasthopesoftware.resources.RecordingApplicationMessageBus
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test

class WhenBroadcastingTheEngineType {

	private val recordingApplicationMessageBus = RecordingApplicationMessageBus()

	@BeforeAll
	fun act() {
		PlaybackEngineTypeChangedBroadcaster(recordingApplicationMessageBus)
			.broadcastPlaybackEngineTypeChanged(PlaybackEngineType.ExoPlayer)
	}

	@Test
	fun `then the exo player selection is broadcast`() {
		assertThat(
			recordingApplicationMessageBus.recordedMessages.filterIsInstance<PlaybackEngineTypeChangedBroadcaster.PlaybackEngineTypeChanged>()
				.firstOrNull()
				?.playbackEngineType
		).isEqualTo(PlaybackEngineType.ExoPlayer)
	}
}
